### PR TITLE
Replace anonymous class with method reference

### PR DESCRIPTION
**Pull request by igor-suhorukov**
## What is the purpose of the change

XXXXX

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by carryxyh** [comment]
@tswstarplanet 
Code in master doesn't support jdk6 or 7

@igor-suhorukov 
LGTM.

File: dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/beans/factory/annotation/AnnotationPropertyValuesAdapterTest.java [hunk 1]
```diff
@@ -66,19 +65,9 @@ public void test() {
...
-            }
-        });
-
-        conversionService.addConverter(new Converter<String[], Map<String, String>>() {
-            @Override
-            public Map<String, String> convert(String[] source) {
-                return CollectionUtils.toStringMap(source);
-            }
-        });
+        conversionService.addConverter((Converter<String[], String>) StringUtils::arrayToCommaDelimitedString);
```
Comment (beiwei30) [inline dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/beans/factory/annotation/AnnotationPropertyValuesAdapterTest.java]: @igor-suhorukov, this change leads to CI failure. It looks like lambda lost generic info (actualTypeArguments).

pls. check org.springframework.core.SerializableTypeWrapper#forGenericInterfaces

```java
	public static Type[] forGenericInterfaces(final Class<?> type) {
		Type[] result = new Type[type.getGenericInterfaces().length];
		for (int i = 0; i < result.length; i++) {
			final int index = i;
			result[i] = forTypeProvider(new SimpleTypeProvider() {
				@Override
				public Type getType() {
					return type.getGenericInterfaces()[index];
				}
			});
		}
		return result;
	}
```

Comment (igor-suhorukov) [reply dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/beans/factory/annotation/AnnotationPropertyValuesAdapterTest.java]: @beiwei30 thank you for assistance

**Comment by beiwei30** [comment]
LGTM now, thanks :)

**Pull request merged**